packages/components/src/focal-point-picker/styles/focal-point-picker-style.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import styled from '@emotion/styled';
+
+/**
+ * Internal dependencies
+ */
+import Flex from '../../flex';
+import BaseUnitControl from '../../unit-control';
+import { color } from '../../utils/style-mixins';
+
+export const MediaWrapper = styled.div`
+	background-color: transparent;
+	box-sizing: border-box;
+	text-align: center;
+	width: 100%;
+`;
+
+export const MediaContainer = styled.div`
+	align-items: center;
+	box-sizing: border-box;
+	box-shadow: 0 0 0 1px rgba( 0, 0, 0, 0.2 );
+	cursor: pointer;
+	display: inline-flex;
+	justify-content: center;
+	margin: auto;
+	position: relative;
+	height: 100%;
+
+	img,
+	video {
+		box-sizing: border-box;
+		display: block;
+		height: auto;
+		margin: 0;
+		max-height: 100%;
+		max-width: 100%;
+		pointer-events: none;
+		user-select: none;
+		width: auto;
+	}
+`;
+
+export const MediaPlaceholder = styled.div`
+	background: ${ color( 'lightGray.300' ) };
+	box-sizing: border-box;
+	height: 170px;
+	max-width: 280px;
+	min-width: 200px;
+	width: 100%;
+`;
+
+export const UnitControl = styled( BaseUnitControl )`
+	width: 100px;
+`;
```

Comment:
So for my own knowledge, this means the sub-styles are passed as a `style` prop right? Are there any other magic-tricks done here? 

Also, this seems to force us to add  a `style` prop for most of our components. How do you see React native support fitting here. One principle we had for some time in `@wordpress/components` is that the public API for our components should  be as DOM agnostic as possible.

Reply:
`styled()` generated CSS are passed as classNames ✌️ .

```
styled.div`...`

// or

styled(SomeComponent)`...`
```

Creates...

```
<div className="css-12xjx1" />
```

With the generated CSS injected (and managed) as:

```
<style>
    ...
    .css-12xjx1 {...}
    ...
</style>
```

Hope this helps!

Reply:
Oh cool! How does this translate in react native?

Reply:
I believe it generates a React Native `Stylesheet` under the hood:

https://emotion.sh/docs/@emotion/native

https://github.com/emotion-js/emotion/blob/master/packages/native/src/styled.js#L7

I've used `styled` from `@emotion/native` in a pure React Native project. It worked great! However, I've personally not worked on project that required the exact same (styled) component to be compatible for both Web and Native.

That would be an interesting challenge!

Reply:
Would be cool to see if these new components work well for the mobile team.